### PR TITLE
excel.selectrange command added.

### PR DESCRIPTION
**Pull request by MarekKosinski**

### CODE REVIEW

**Review by pwitusik — APPROVED** [review]

File: G1ANT.Addon.MSOffice/Commands/ExcelSelectRangeCommand.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        {
+            try
+            {
+                object col1, col2;
+                if (arguments.ColIndex1 != null)
+                    col1 = arguments.ColIndex1.Value;
+                else if (arguments.ColName1 != null)
+                    col1 = arguments.ColName1.Value;
+                else
+                    throw new ArgumentException("One of the ColIndex1 or ColName1 arguments have to be set up.");
```
Comment (Vegebutcher) [inline G1ANT.Addon.MSOffice/Commands/ExcelSelectRangeCommand.cs]:
```suggestion
                    throw new ArgumentException("One of the ColIndex1 or ColName1 arguments has to be set up.");
```

File: G1ANT.Addon.MSOffice/Commands/ExcelSelectRangeCommand.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+                if (arguments.ColIndex2 != null)
+                    col2 = arguments.ColIndex2.Value;
+                else if (arguments.ColName2 != null)
+                    col2 = arguments.ColName2.Value;
+                else
+                    throw new ArgumentException("One of the ColIndex2 or ColName2 arguments have to be set up.");
```
Comment (Vegebutcher) [inline G1ANT.Addon.MSOffice/Commands/ExcelSelectRangeCommand.cs]:
```suggestion
                    throw new ArgumentException("One of the ColIndex2 or ColName2 arguments has to be set up.");
```

File: G1ANT.Addon.MSOffice/Commands/ExcelSelectRangeCommand.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+
+                ExcelManager.CurrentExcel.SelectRange(col1, arguments.Row1.Value, col2, arguments.Row2.Value);
+            }
+            catch (Exception ex)
+            {
+                throw new ApplicationException($"Error occured while selecting range in current excel instance. Message: '{ex.Message}'", ex);
```
Comment (Vegebutcher) [inline G1ANT.Addon.MSOffice/Commands/ExcelSelectRangeCommand.cs]:
```suggestion
                throw new ApplicationException($"Error occurred while selecting a range in the current Excel instance. Message: '{ex.Message}'", ex);
```

File: G1ANT.Addon.MSOffice/Api/ExcelWrapper.cs [hunk 4]
```diff
@@ -238,6 +238,19 @@ public void Close()
             }
         }
 
+        public void SelectRange(object startColumn, int startRow, object endColumn, int endRow)
+        {
+            if (startColumn == null || endColumn == null)
+            {
+                throw new ApplicationException("Starting cell's column and ending cell's column needs to be specified.");
```
Comment (Vegebutcher) [inline G1ANT.Addon.MSOffice/Api/ExcelWrapper.cs]:
```suggestion
                throw new ApplicationException("Starting cell's column and ending cell's column need to be specified.");
```

File: G1ANT.Addon.MSOffice/Commands/ExcelSelectRangeCommand.cs [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+using Org.BouncyCastle.Crypto.Agreement.JPake;
+
+namespace G1ANT.Addon.MSOffice
+{
+    [Command(Name = "excel.selectrange", Tooltip = "This command selects a range in the currently active Excel instance")]
+    public class ExcelSelectRangeCommand : Command
+    {
+        public class Arguments : CommandArguments
+        {
+            [Argument(Tooltip = "Starting Cell's column index")]
```
Comment (danieladamkaiser) [inline G1ANT.Addon.MSOffice/Commands/ExcelSelectRangeCommand.cs:21]: Add tags 
```
[Argument(Required = true...()]
```
to required fields

**Review by danieladamkaiser — APPROVED** [review]

**Pull request merged**